### ISSUE TITLE
Secondary file imports

### ISSUE THREAD

**Issue by denis-yuen** [ga4gh/dockstore#1397]
### Desired behaviour

Branches from #1041 

See if we can find a temporary solution to identifying and indexing secondary files for Nextflow. 
It looks like currently, it is possible to re-use Groovy scripts via the standard Groovy import mechanism. 
Examples include 
https://github.com/guigolab/ipsa-nf/blob/master/ipsa.nf#L26 
https://github.com/SciLifeLab/Sarek/blob/master/main.nf#L60 

Note, it looks like the NextFlow equivalent of subworkflows does not exist yet https://github.com/nextflow-io/nextflow/issues/238 (so we don't need to worry about the equivalent of CWL's "run" field *https://www.commonwl.org/v1.0/Workflow.html#WorkflowStep )

**Comment by pditommaso** [comment]
Nextflow syntax is a superset of Groovy. Therefore a nextflow script (and config file) is able to compile groovy (and java) code ie. imports, class definition, etc., tho this is not a suggested implementation best practice. 

> if we can find a temporary solution to identifying and indexing secondary files for Nextflow

Is this needed to identify the workflows input files?

**Comment by denis-yuen** [comment]
@pditommaso 
Workflow input files is part of it, long term we have the ambition of understanding what workflows have what input files and in what formats. 

Short term, for WDL and CWL, we like to display all relevant secondary descriptors (via imports) so users can browse all the files used in a workflow (and Dockstore can potentially pass them along via TRS). So I'm currently looking through the `bin` and `lib` directories for a workflow for possible imports.

**Comment by denis-yuen** [comment]
@pditommaso 

Ah right, sorry I was being unclear. Yes, I meant secondary files as in files imported into the descriptor as opposed to CWL's concept of secondary input files. 

Can we assume that workflow steps (I think they're called processes in Nextflow) are all contained within the main workflow script? (or can files in the `lib/` folder declare new processes)?

**Comment by pditommaso** [comment]
> Can we assume that workflow steps (I think they're called processes in Nextflow) are all contained within the main workflow script?

Yes, a NF workflow is a self-contained script. It can executed external scripts, but all processes (ie. steps) are defined in the workflow script. 

Although the config file can import other config files. Looks for example [here](https://github.com/nf-core/methylseq/blob/master/nextflow.config#L46-L67).
